odl/tomo/geometry/geometry.py

```diff
@@ -164,26 +164,6 @@ def det_refpoint(self, mpar):
             The reference point, an `ndim`-dimensional vector
         """
 
-    @abstractmethod
```

Comment:
In principle that's true. It would be possible to make the hierarchy a bit more fine-grained (perhaps by duck-typing) by differentiating between geometries where the detector moves relative to the object (CT, SPECT) and ones where it does not (PET, MRI, anything with an array of fixed detectors).

The case with the rotation matrix (i.e. a rotating detector in general) could encompass all cases where the detector movement can be parametrised in the in polar coordinates `x = r(phi) * d(phi)`, where `r(phi)` is the distance from a reference point (e.g. 0) and `d(phi)` is a direction vector. The circular acquisition curves would be special cases of that, but we also need more general cases e.g. in SPECT, where `r(phi)` is given as a list of numbers. Probably the detector movement is determined during acquisition so I don't know if the points will lie on any reasonably regular curve.

The axis-oriented case is similar, except that the polar coordinates only apply perpendicular to the axis, while a third function `g(phi)` governs the movement along the axis.

Anyway, this is probably better suited for a separate issue. The removal of the matrix was anyway motivated by the imminent PET geometry implementation, which took a turn towards SPECT in the end, so we can keep the matrix in this PR.